internal/pkg/ocisif/overlay.go

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2024, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package ocisif
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	ocitsif "github.com/sylabs/oci-tools/pkg/sif"
+	"github.com/sylabs/sif/v2/pkg/sif"
+	"github.com/sylabs/singularity/v4/pkg/image"
+)
+
+var Ext3LayerMediaType types.MediaType = "application/vnd.sylabs.image.layer.v1.ext3"
+
+// HasOverlay returns whether the OCI-SIF at imgPath has an ext3 writable final
+// layer - an 'overlay'.
+func HasOverlay(imagePath string) (bool, error) {
+	fi, err := sif.LoadContainerFromPath(imagePath,
+		sif.OptLoadWithFlag(os.O_RDONLY),
+		sif.OptLoadWithCloseOnUnload(false),
```

Comment:
Think you probably _do_ want to close the backing file on `UnloadContainer`, so the file handle doesn't leak?

```suggestion
```

Reply:
Yikes.. thanks... copied and pasted that from somewhere 😬